test/tests/functional/pool_tests.js

```diff
@@ -39,6 +39,36 @@ describe('Pool tests', function() {
     }
   });
 
+  it('Should only listen on connect once', {
+    metadata: { requires: { topology: 'single' } },
+
+    test: function(done) {
+      // Enable connections accounting
+      Connection.enableConnectionAccounting();
+
+      // Attempt to connect
+      var pool = new Pool(null, {
+        host: this.configuration.host,
+        port: this.configuration.port,
+        bson: new Bson(),
+        messageHandler: function() {}
+      });
+
+      // Add event listeners
+      pool.on('connect', function(_pool) {
+        var connections = _pool.allConnections();
+        expect(connections).to.have.lengthOf(1).and.to.have;
+        expect(connections[0].connection.listenerCount('connect')).to.equal(1);
+        _pool.destroy();
+        Connection.disableConnectionAccounting();
```

Comment:
I think what we actually want to test is that there is a single listener count until `connect` occurs right? So there should be an initial listener count for that event, and when connect occurs there should be zero listeners.  Or am I misunderstanding something?

Reply:
I'd expect the listener count to be 0 as well, but there is some listener nestled deep inside of node's core that we don't have control over.

Still, I will refactor to emphasize that the number of listeners for the connection should decrease by 1.